data-transformers/transformers/propertiesTransformer.py

```diff
@@ -189,74 +192,72 @@ def createVersionDict(a):
             continue
     return d
 
+def getFromMinio(url):
+    from commons import getFromUrl
+    return getFromUrl(url)
+
+def hostNeedsSaving(infoData, cassandraKeyspace):
+    hostID = ""
+    
+    ob = json.loads(infoData.decode())
+    for e in ob:
+        e = e.split("=")
+        if e[0] == "ID":
+            hostID = e[1]
+            break
+    
+    if hostID == "":
+        return false
+    
+    hostNotSaved = sc.cassandraTable(cassandraKeyspace, "host_properties") \
+        .select("host_id") \
+        .where("host_id=?", hostID) \
+        .isEmpty()
+        
+    return hostNotSaved
+
 def main():
     from pyspark_cassandra import CassandraSparkContext
     from pyspark import SparkConf
+    from pyspark import SparkFiles
     
     # Takes arguments
-    sparkMaster = sys.argv[1]
-    cassandraHost = sys.argv[2]
-    minioHost = sys.argv[3]
-    filePath = sys.argv[4]
-    trialID = sys.argv[5]
-    experimentID = trialID.split("_")[0]
-    containerID = sys.argv[7]
-    cassandraKeyspace = "benchflow"
+    minioHost = sys.argv[1]
+    filePath = sys.argv[2]
+    trialID = sys.argv[3]
+    experimentID = sys.argv[4]
+    SUTName = sys.argv[5]
+    containerID = sys.argv[6]
     table = "container_properties"
-    minioPort = "9000"
     
     # Set configuration for spark context
-    conf = SparkConf() \
-        .setAppName("Properties Transformer") \
-        .setMaster(sparkMaster) \
-        .set("spark.cassandra.connection.host", cassandraHost)
+    conf = SparkConf().setAppName("Properties Transformer")
     sc = CassandraSparkContext(conf=conf)
     
-    res = urllib2.urlopen("http://"+minioHost+":"+minioPort+"/"+filePath+"_inspect.gz")
-    compressed = io.BytesIO(res.read())
-    decompressed = gzip.GzipFile(fileobj=compressed)
-    inspectData = decompressed.readline()
+    # Retrieve the configuration file that was sent to spark
+    confPath = SparkFiles.get("data-transformers.yml")
+    with open(confPath) as f:
+        transformerConfiguration = yaml.load(f)
+        cassandraKeyspace = transformerConfiguration["cassandra_keyspace"]
+        minioPort = transformerConfiguration["minio_port"]
     
+    inspectData = getFromMinio("http://"+minioHost+":"+minioPort+"/"+filePath+"_inspect.gz")[0]  
```

Comment:
@VincenzoFerme It's a list containing all the lines in the file. But these files only have one line, so I just take that